frontend/src/chain/tx/utils/getPreSigningHashes.ts

```diff
@@ -25,34 +43,25 @@ export const getPreSigningHashes = ({
     txInputData
   );
 
-  if (isOneOf(chain, Object.values(UtxoChain))) {
-    const { errorMessage, hashPublicKeys } =
-      TW.Bitcoin.Proto.PreSigningOutput.decode(preHashes);
+  const chainKind = chainKindRecord[chain];
 
-    assertErrorMessage(errorMessage);
+  const decoder = decoders[chainKind];
 
-    return withoutNullOrUndefined(hashPublicKeys.map(hash => hash?.dataHash));
-  } else if (chain === Chain.Solana) {
-    const { errorMessage, data } =
-      TW.Solana.Proto.PreSigningOutput.decode(preHashes);
+  const output = decoder(preHashes);
 
-    assertErrorMessage(errorMessage);
+  assertErrorMessage(output.errorMessage);
 
-    return [data];
+  if ('hashPublicKeys' in output) {
+    return withoutNullOrUndefined(
+      output.hashPublicKeys.map(hash => hash?.dataHash)
+    );
   }
 
-  const { errorMessage, dataHash, data } =
-    TW.TxCompiler.Proto.PreSigningOutput.decode(preHashes);
-
-  assertErrorMessage(errorMessage);
-
-  if (dataHash.length === 0) {
-    if (chain === Chain.Sui) {
-      return [walletCore.Hash.blake2b(data, 32)];
```

Comment:
@rcoderdev this cahnge will cause issue with `SUI`, correct?

Reply:
No, I checked, `dataHash` is the same as `walletCore.Hash.blake2b(data, 32)`